### ISSUE TITLE
`--help` doesn't show optional flag  `--all`

### ISSUE THREAD

**Issue by heiseish** [rust-lang/rust-clippy#14639]
### Summary

Was going through the command-line to make sure I should use `--all` or `--workspace`, but it seems neither option is highlighted in `--help`

```bash
❯ cargo clippy --help
Checks a package to catch common mistakes and improve your Rust code.

Usage:
    cargo clippy [OPTIONS] [--] [<ARGS>...]

Common options:
    --no-deps                Run Clippy only on the given crate, without linting the dependencies
    --fix                    Automatically apply lint suggestions. This flag implies --no-deps and --all-targets
    -h, --help               Print this message
    -V, --version            Print version info and exit
    --explain [LINT]         Print the documentation for a given lint

See all options with cargo check --help.

Allowing / Denying lints

To allow or deny a lint from the command line you can use cargo clippy -- with:

    -W / --warn [LINT]       Set lint warnings
    -A / --allow [LINT]      Set lint allowed
    -D / --deny [LINT]       Set lint denied
    -F / --forbid [LINT]     Set lint forbidden

You can use tool lints to allow or deny lints from your code, e.g.:

    #[allow(clippy::needless_lifetimes)]

Manifest Options:
    --manifest-path <PATH>  Path to Cargo.toml
    --frozen                Require Cargo.lock and cache are up to date
    --locked                Require Cargo.lock is up to date
    --offline               Run without accessing the network
```

Not sure if this is intended. If not, I'm happy to open a PR if someone could give a quick pointer.

### Reproducer

I tried this code:

```bash
cargo clippy --help
```

I expected to see this happen:
`--all` to show up
Instead, this happened:
None

### Version

```bash
❯ cargo -V clippy
cargo 1.88.0-nightly (864f74d4e 2025-04-11)
```

**Comment by MCJOHN974** [comment]
The command you posted suggest's you to run `cargo check --help`:

 `See all options with cargo check --help.`

And `cargo check --help` tells about `--all` flag.
